answerkey/datastructures/24.answer.scala

```diff
@@ -16,8 +16,10 @@ def startsWith[A](l: List[A], prefix: List[A]): Boolean = (l,prefix) match {
   case _ => false
 }
 @annotation.tailrec
-def hasSubsequence[A](sup: List[A], sub: List[A]): Boolean = sup match {
-  case Nil => false
-  case _ if startsWith(sup, sub) => true
-  case Cons(h,t) => hasSubsequence(t, sub)
+def hasSubsequence[A](l: List[A], sub: List[A]): Boolean = (l, sub) match {
+  case (Nil, Nil) => true
+  case (Nil, _) => false
+  case (_, Nil) => false
```

Comment:
The empty string should be a substring of any string. This case should return true.